core/src/main/java/org/pentaho/di/core/vfs/KettleVFS.java

```diff
@@ -132,71 +131,73 @@ public static FileObject getFileObject( String vfsFilename, FileSystemOptions fs
   // However, how are we going to verify this?
   //
   // We are going to see if the filename starts with one of the known protocols like file: zip: ram: smb: jar: etc.
-  // If not, we are going to assume it's a file, marking the relativeFilename flag as true, and it only changes if
+  // If not, we are going to assume it's a file, when no scheme found ( flag as null ), and it only changes if
   // a scheme is provided.
   //
   public static FileObject getFileObject( String vfsFilename, VariableSpace space, FileSystemOptions fsOptions )
     throws KettleFileException {
+
+    //  Protect the code below from invalid input.
+    if ( vfsFilename == null ) {
+      throw new KettleFileException( "Unexpected null VFS filename",
+        new NullPointerException( "KettleVFS.getFileObject() vfsFilename is NULL" ) );
```

Comment:
I believe IllegalArgumentException is more apropriate.
Also, Why the need to encapsulate the exception in a KettleFileException?

Reply:
Changed.